File: leetcode/2130. Maximum Twin Sum of a Linked List.dart
import 'dart:ffi';
import 'dart:math';

/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class ListNode {
  int val;
  ListNode? next;
  ListNode([this.val = 0, this.next]);
}

class Solution {
  int pairSum(ListNode? head) {
    List vals = new List.empty(growable: true);
    while (head != null) {
      vals.add(head.val);
      head = head.next;
    }
    int ans = -1;
    for (int i = 0; i < vals.length / 2; i++) {
      ans = max(ans, vals[i] + vals[vals.length - i - 1]);
    }
    return ans;
  }
}

File: leetcode/2130. Maximum Twin Sum of a Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var pairSum = function(head) {
    let vals = new Array;
    while (head){
        vals.push(head.val);
        head = head.next;
    }
    let ans = -1;
    for (let i=0; i<vals.length/2; i++){
        ans = ans > vals[i]+vals[vals.length-i-1] ? ans: vals[i]+vals[vals.length-i-1]
    }
    return ans;
};